pylana/aggregation.py

```diff
@@ -0,0 +1,161 @@
+            activity_exclusion_filter:
+                A list containing the activities to exclude.
+            value_sorting:
+                A string denoting the metric to sort the aggregation by.
+            sorting_order:
+                A string denoting the order of the sorting.
+            values_from:
+                A string denoting which values to consider for the aggregation.
+            aggregation_function:
+                A string denoting the aggregation function to use.
+                Can be one of ["min", "max", "sum", "mean", "median", "variance",
+                "standardDeviation"] or a percentile as a string starting with "p"
+                followed by the percentile value.
+            date_type:
+                A string denoting the date type of the grouping.
+            secondary_date_type:
+                A string denoting the date type of the secondary grouping.
+            **kwargs:
+                Keyword arguments passed to requests functions.
+
+        Returns:
+            A pandas DataFrame containing the aggregated data.
+        """
+
+        request_data = {'metric': create_metric(metric, aggregation_function),
+                        'valuesFrom': {'type': values_from},
+                        'miningRequest': {'logId': log_id,
+                                          'activityExclusionFilter': activity_exclusion_filter,
+                                          'traceFilterSequence': trace_filter_sequence},
+                        'options': {'maxAmountAttributes': max_amount_attributes,
+                                    'valueSorting': value_sorting,
+                                    'sortingOrder': sorting_order}}
+
+        if grouping is not None:
+            request_data['grouping'] = create_grouping(grouping, date_type)
+
+        if secondary_grouping is not None:
+            request_data['secondaryGrouping'] = create_grouping(secondary_grouping, secondary_date_type)
+
+        aggregate_response = self.post('/api/v2/aggregate-data', json = request_data, **kwargs)
+
+        if aggregate_response.status_code >= 400:
+            return pd.DataFrame()
+
+        response_df = pd.DataFrame(aggregate_response.json()['chartValues'])
+
+        if secondary_grouping is not None:
+            response_df = response_df.explode('values').reset_index(drop = True)
+
+            z_axis = response_df['zAxis']
+
+            response_df = json_normalize(response_df['values'])
+
+            response_df['zAxis'] = z_axis
+
+        response_df.drop('$type', axis = 1, inplace = True)
+
+        response_df = response_df.rename(columns = {'xAxis': grouping,
+                                                    'yAxis': metric,
+                                                    'zAxis': secondary_grouping})
+        return(response_df)
+
+
+    def boxplot_stats(self, log_id: str, metric: str, grouping: str = None,
```

Comment:
We should consider not referencing a boxplot, since it not well defined which quantities are displayed. Some people denote outliers as values outside the 5% and 95% percentiles. I would rather call it `describe` as the pandas data frame method. I would orient at the outcome of that method since it is very instructive and already known to users (one new thing less to learn).

Also, we should describe what we are expecting to get in the doc string (which location parameters, etc.). If we know that, we should also mention whether approximations are used for the percentile calculation. But that's maybe something for the future.

Reply:
So I would say: yes to better docstrings, but no to calling it describe, since these are the default boxplot stats (used in 98% of when you need a boxplot). I updated the docstring accordingly.

Reply:
Agreed.